File: Scripts/results/generated_code_results/baseline_claude/Education/OnlineTeachingSoftware/CourseDetailScreen.js
import React from 'react';

import { View, Text, Image, StyleSheet, TouchableOpacity, ScrollView } from 'react-native';

import { useNavigation } from '@react-navigation/native';

import Icon from 'react-native-vector-icons/Ionicons';



const CourseDetailScreen = () => {

  const navigation = useNavigation();



  return (

    <ScrollView style={styles.container}>

      <View style={styles.header}>

        <TouchableOpacity onPress={() => navigation.goBack()}>

          <Icon name="arrow-back" size={24} color="#000" />

        </TouchableOpacity>

        <TouchableOpacity>

          <Icon name="ellipsis-vertical" size={24} color="#000" />

        </TouchableOpacity>

      </View>

      

      <View style={styles.imageContainer}>

        <Image

          source={require('../assets/snack-icon.png')}

          style={styles.courseImage}

        />

        <TouchableOpacity style={styles.favoriteButton}>

          <Icon name="heart-outline" size={24} color="#000" />

        </TouchableOpacity>

      </View>



      <View style={styles.contentContainer}>

        <Text style={styles.tag}>UI Design</Text>

        <Text style={styles.title}>Expert Wireframing for Webisite Design theme food</Text>

        

        <View style={styles.instructorContainer}>

          <Image

            source={require('../assets/snack-icon.png')}

            style={styles.instructorImage}

          />

          <Text style={styles.instructorName}>Jerremy Mamika</Text>

          <View style={styles.ratingContainer}>

            <Icon name="star" size={16} color="#FFD700" />

            <Text style={styles.rating}>4.6</Text>

          </View>

        </View>



        <Text style={styles.description}>

          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s,

        </Text>



        <View style={styles.statsContainer}>

          <View style={styles.statItem}>

            <Text style={styles.statValue}>30+</Text>

            <Text style={styles.statLabel}>Lessons</Text>

          </View>

          <View style={styles.statItem}>

            <Text style={styles.statValue}>4.5</Text>

            <Text style={styles.statLabel}>Rating</Text>

          </View>

          <View style={styles.statItem}>

            <Text style={styles.statValue}>1.4K</Text>

            <Text style={styles.statLabel}>Students</Text>

          </View>

        </View>



        <View style={styles.priceContainer}>

          <Text style={styles.price}>$129</Text>

          <TouchableOpacity style={styles.buyButton}>

            <Text style={styles.buyButtonText}>Buy Now</Text>

          </TouchableOpacity>

        </View>

      </View>

    </ScrollView>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#fff',

  },

  header: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    padding: 16,

  },

  imageContainer: {

    position: 'relative',

  },

  courseImage: {

    width: '100%',

    height: 200,

    resizeMode: 'cover',

  },

  favoriteButton: {

    position: 'absolute',

    top: 16,

    right: 16,

    backgroundColor: '#fff',

    borderRadius: 20,

    padding: 8,

  },

  contentContainer: {

    padding: 16,

  },

  tag: {

    backgroundColor: '#F0F0F0',

    paddingHorizontal: 8,

    paddingVertical: 4,

    borderRadius: 4,

    alignSelf: 'flex-start',

    marginBottom: 8,

  },

  title: {

    fontSize: 24,

    fontWeight: 'bold',

    marginBottom: 16,

  },

  instructorContainer: {

    flexDirection: 'row',

    alignItems: 'center',

    marginBottom: 16,

  },

  instructorImage: {

    width: 40,

    height: 40,

    borderRadius: 20,

    marginRight: 8,

  },

  instructorName: {

    fontSize: 16,

    fontWeight: '500',

    marginRight: 8,

  },

  ratingContainer: {

    flexDirection: 'row',

    alignItems: 'center',

  },

  rating: {

    marginLeft: 4,

  },

  description: {

    marginBottom: 16,

    lineHeight: 20,

  },

  statsContainer: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    marginBottom: 16,

  },

  statItem: {

    alignItems: 'center',

  },

  statValue: {

    fontSize: 18,

    fontWeight: 'bold',

  },

  statLabel: {

    color: '#666',

  },

  priceContainer: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    alignItems: 'center',

  },

  price: {

    fontSize: 24,

    fontWeight: 'bold',

  },

  buyButton: {

    backgroundColor: '#6C5CE7',

    paddingHorizontal: 24,

    paddingVertical: 12,

    borderRadius: 8,

  },

  buyButtonText: {

    color: '#fff',

    fontWeight: 'bold',

  },

});



export default CourseDetailScreen;

File: Scripts/results/generated_code_results/Flutter/Education/OnlineTeachingSoftware/CourseDetailScreen.dart
import 'package:flutter/material.dart';



class CourseDetailScreen extends StatelessWidget {

  @override

  Widget build(BuildContext context) {

    return Scaffold(

      body: SafeArea(

        child: Column(

          crossAxisAlignment: CrossAxisAlignment.start,

          children: [

            Stack(

              children: [

                Image.asset('assets/images/template.png', fit: BoxFit.cover),

                Positioned(

                  top: 16,

                  left: 16,

                  child: IconButton(

                    icon: Icon(Icons.arrow_back, color: Colors.white),

                    onPressed: () => Navigator.pop(context),

                  ),

                ),

                Positioned(

                  top: 16,

                  right: 16,

                  child: PopupMenuButton(

                    icon: Icon(Icons.more_vert, color: Colors.white),

                    itemBuilder: (context) => [

                      PopupMenuItem(child: Text('Option 1')),

                      PopupMenuItem(child: Text('Option 2')),

                    ],

                  ),

                ),

                Positioned(

                  top: 16,

                  left: 16,

                  child: Container(

                    padding: EdgeInsets.symmetric(horizontal: 8, vertical: 4),

                    decoration: BoxDecoration(

                      color: Colors.white,

                      borderRadius: BorderRadius.circular(16),

                    ),

                    child: Text('UI Design', style: TextStyle(color: Colors.black)),

                  ),

                ),

                Positioned(

                  bottom: 16,

                  right: 16,

                  child: IconButton(

                    icon: Icon(Icons.favorite_border, color: Colors.white),

                    onPressed: () {},

                  ),

                ),

              ],

            ),

            Padding(

              padding: const EdgeInsets.all(16.0),

              child: Column(

                crossAxisAlignment: CrossAxisAlignment.start,

                children: [

                  Text(

                    'Expert Wireframing for Webiste Design theme food',

                    style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),

                  ),

                  SizedBox(height: 16),

                  ListTile(

                    contentPadding: EdgeInsets.zero,

                    leading: CircleAvatar(

                      backgroundImage: AssetImage('assets/images/template.png'),

                    ),

                    title: Text('Jerremy Mamika'),

                    trailing: Row(

                      mainAxisSize: MainAxisSize.min,

                      children: [

                        Icon(Icons.star, color: Colors.yellow),

                        Text('4.6'),

                      ],

                    ),

                  ),

                  SizedBox(height: 16),

                  Text(

                    'About Course',

                    style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),

                  ),

                  SizedBox(height: 8),

                  Text(

                    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s,",

                  ),

                  SizedBox(height: 16),

                  Row(

                    mainAxisAlignment: MainAxisAlignment.spaceBetween,

                    children: [

                      _buildStatItem('30+', 'Lessons'),

                      _buildStatItem('4.5', 'Rating'),

                      _buildStatItem('1.4K', 'Student'),

                    ],

                  ),

                  SizedBox(height: 16),

                  Row(

                    mainAxisAlignment: MainAxisAlignment.spaceBetween,

                    children: [

                      Column(

                        crossAxisAlignment: CrossAxisAlignment.start,

                        children: [

                          Text('Price', style: TextStyle(color: Colors.grey)),

                          Text('\$129', style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold)),

                        ],

                      ),

                      ElevatedButton(

                        onPressed: () {},

                        child: Text('Buy Now'),

                        style: ElevatedButton.styleFrom(

                          backgroundColor: Colors.purple,

                          padding: EdgeInsets.symmetric(horizontal: 32, vertical: 16),

                        ),

                      ),

                    ],

                  ),

                ],

              ),

            ),

          ],

        ),

      ),

    );

  }



  Widget _buildStatItem(String value, String label) {

    return Column(

      children: [

        Text(value, style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),

        Text(label, style: TextStyle(color: Colors.grey)),

      ],

    );

  }

}